### ISSUE TITLE
When shutting down Edge, we should persist metrics upstream.

### ISSUE THREAD

**Issue by chriswk** [Unleash/unleash-edge#294]
### Describe the feature request

I would like Edge to persist metrics upstream on shutdown.

### Background

Currently, if Edge has persistence configured, we do already persist tokens/features to the configured persistence when shutting down. We should also make one last attempt to upload metrics to upstream when shutting down.

**Comment by ivarconr** [comment]
Yes, flushing cached metrics would be preferred on shutdown. This is in line with how node SDK (and thus the proxy) do it today.

**Comment by sighphyre** [comment]
Closed by #564
